graphql/src/Mu/GraphQL/Quasi.hs

```diff
@@ -0,0 +1,154 @@
+    argToType (GQL.InputValueDefinition _ aname _ _) =
+      [t| 'ArgSingle 'Nothing '[] ('SchemaRef $(conT schemaName) $(textToStrLit (GQL.unName aname))) |]
+    retToType :: GQL.GType -> Q Type
+    retToType (GQL.TypeNamed _ a) =
+      [t| 'RetSingle ('SchemaRef $(conT schemaName) $(textToStrLit . GQL.unName . GQL.unNamedType $ a)) |]
+    retToType _ = fail "only named types may be used as results"
+typeToDec _ (GQL.TypeDefinitionInterface _)       = fail "interface types are not supported"
+typeToDec _ (GQL.TypeDefinitionUnion _)           = fail "union types are not supported"
+typeToDec _ (GQL.TypeDefinitionEnum enums)        = enumToDecl enums
+  where
+    enumToDecl :: GQL.EnumTypeDefinition -> Q Result
+    enumToDecl = error "not implemented" -- TODO: this will return a `GQLSchema`
+typeToDec _ (GQL.TypeDefinitionInputObject inpts) = inputObjToDec inpts
+  where
+    inputObjToDec :: GQL.InputObjectTypeDefinition -> Q Result
+    inputObjToDec = error "not implemented" -- TODO: this will return a `GQLSchema`
+
+-- data ObjectTypeDefinition
+--   = ObjectTypeDefinition
+--   { _otdDescription          :: !(Maybe Description)
+--   , _otdName                 :: !Name
+--   , _otdImplementsInterfaces :: ![NamedType]
+--   , _otdDirectives           :: ![Directive]
+--   , _otdFieldsDefinition     :: ![FieldDefinition]
+--   }
+
+-- data FieldDefinition
+--   = FieldDefinition
+--   { _fldDescription         :: !(Maybe Description)
+--   , _fldName                :: !Name
+--   , _fldArgumentsDefinition :: !ArgumentsDefinition
+--   , _fldType                :: !GType
+--   , _fldDirectives          :: ![Directive]
+--   }
+
+-- type ArgumentsDefinition = [InputValueDefinition]
+
+-- data InputValueDefinition
+--   = InputValueDefinition
+--   { _ivdDescription  :: !(Maybe Description)
+--   , _ivdName         :: !Name
+--   , _ivdType         :: !GType
+--   , _ivdDefaultValue :: !(Maybe DefaultValue)
+--   }
+
+-- data GType
+--   = TypeNamed !Nullability !NamedType
+--   | TypeList !Nullability !ListType
+
+-- newtype Nullability
+--   = Nullability { unNullability :: Bool }
+
+-- newtype NamedType
+--   = NamedType { unNamedType :: Name }
+
+-- newtype ListType
+--   = ListType {unListType :: GType }
+
+-- data EnumTypeDefinition
+--   = EnumTypeDefinition
+--   { _etdDescription      :: !(Maybe Description)
+--   , _etdName             :: !Name
+--   , _etdDirectives       :: ![Directive]
+--   , _etdValueDefinitions :: ![EnumValueDefinition]
+--   }
+
+-- data InputObjectTypeDefinition
+--   = InputObjectTypeDefinition
+--   { _iotdDescription      :: !(Maybe Description)
+--   , _iotdName             :: !Name
+--   , _iotdDirectives       :: ![Directive]
+--   , _iotdValueDefinitions :: ![InputValueDefinition]
+--   }
+
+schemaFromGQL :: [GQL.ValueConst] -> Q Type
+schemaFromGQL = (typesToList <$>) . traverse schemaFromGQLValue
+
+schemaFromGQLValue :: GQL.ValueConst -> Q Type
```

Comment:
This is not useful at all. You will need to eventually write a function with that signature, but not to return the type, but to handle default values. For now on, you can just remove it.